front/src/pods/create-session/create-session.component.tsx

```diff
@@ -1,41 +1,52 @@
 import React from 'react';
 import { cx } from 'emotion';
 import * as classes from './create-session.styles';
-import { Typography } from '@material-ui/core';
-import TextField from '@material-ui/core/TextField';
-import Button from '@material-ui/core/Button';
+import { Formik, Form } from 'formik';
+import { Typography, Button } from '@material-ui/core';
+import { TextFieldComponent } from 'common/components/forms';
+import { formValidation } from './create-session.validation';
 
 interface Props {
   onCreateSession: (nickname: string) => void;
 }
 
 export const CreateSessionComponent: React.FunctionComponent<Props> = props => {
   const { onCreateSession } = props;
-  const [nickname, setNickname] = React.useState('master');
 
   return (
     <>
       <div className={classes.container}>
         <Typography className={classes.title} variant="h5">
           Enter your name and click on create session
         </Typography>
-        <div className={classes.formContainer}>
-          <TextField
-            className={cx(classes.formItem, classes.textField)}
-            label="Nickname"
-            margin="normal"
-            value={nickname}
-            onChange={e => setNickname(e.target.value)}
-          />
-          <Button
-            className={cx(classes.formItem, classes.button)}
-            variant="contained"
-            color="primary"
-            onClick={e => onCreateSession(nickname)}
-          >
-            Create new session
-          </Button>
-        </div>
+        <Formik
+          onSubmit={sessionForm => onCreateSession(sessionForm.nickname)}
+          initialValues={{ nickname: 'master of puppets' }}
+          validate={formValidation.validateForm}
+        >
+          {props => {
+            const { handleChange, values } = props;
+            return (
+              <Form className={classes.formContainer}>
+                <TextFieldComponent
+                  name="nickname"
+                  label="Nickname"
+                  value={values.nickname}
+                  onChange={handleChange}
```

Comment:
Since this is managed by formik we don't need to inform "value" and "onChange" it should remain like:

```diff
<TextFieldComponent
                  name="nickname"
                  label="Nickname"
-                 value={values.nickname}
-                  onChange={handleChange}
className={cx(classes.formItem, classes.textField)}
                />
```

Will check maybe will introduce my self the update directly